ci/pkl-pipelines/pay-dev/pr-ci.pkl

```diff
@@ -0,0 +1,257 @@
+    name = "pay-ci-endtoend-config-pull-request" 
+    repo = "pay-ci"
+    source { ["paths"] = new Listing<String> { "ci/tasks/endtoend/" } }
+  }
+  new PayResources.PayGitHubPullRequestResource {
+    name = "ci-pull-request" 
+    repo = "pay-ci"
+    source { ["paths"] = new Listing<String> { "ci/pipelines/*"  "ci/tasks/*"} }
+  }
+  new Pipeline.Resource {
+    name = "pull-request-builds-ecr"
+    icon = "docker"
+    type = "registry-image"
+    source = new {
+      ["aws_access_key_id"] = "((readonly_access_key_id))"
+      ["aws_ecr_registry_id"] = "((pay_aws_test_account_id))"
+      ["aws_region"] = "eu-west-1"
+      ["aws_role_arn"] = "arn:aws:iam::((pay_aws_test_account_id)):role/concourse"
+      ["aws_secret_access_key"] = "((readonly_secret_access_key))"
+      ["aws_session_token"] = "((readonly_session_token))"
+      ["repository"] = "govukpay/pull-request-builds"
+      ["tag"] = "latest"
+    }
+  }
+}
+
+jobs = new {
+  pipeline_self_update.PayPipelineSelfUpdateJob("pay-dev/pr-ci.pkl")
+  for (app, config in e2e_test_apps) { e2eJob(app, config) }
+  new {
+    name = "ci-pr-test"
+    max_in_flight = 3
+    build_log_retention = new { builds = 500 }
+    plan {
+      shared_test.getPayCi
+      pullRequestGetStep("ci-pull-request")
+      putStatus("ci-pull-request", "test", "pending")
+
+      new TaskStep {
+        task = "check-pipelines-and-tasks"
+        file = "pay-ci/ci/tasks/check-pipelines-and-tasks.yml"
+        on_failure = putStatus("ci-pull-request", "test", "failure")
+      }
+      putStatus("ci-pull-request", "test", "success")
+    }
+  }
+  new {
+    name = "pay-ci-endtoend-config"
+    max_in_flight = 3
+    build_log_retention = new { builds = 500 }
+    on_failure = putStatus("pay-ci-endtoend-config-pull-request", "e2e tests", "failure")
+    plan {
+      new InParallelStep {
+        in_parallel = new Listing<Step> {
+          shared_test.getPayCi
+          pullRequestGetStep("pay-ci-endtoend-config-pull-request")
+        }
+      }
+      putStatus("pay-ci-endtoend-config-pull-request", "e2e tests", "pending")
+      shared_test.assumeCodeBuildRole("executor", "pay-ci-endtoend-config-pr-assume-role")
+      shared_test.loadVar("role", "assume-role/assume-role.json")
+      shared_test.prepareCodeBuild("pay-ci", "prepare-e2e-codebuild.yml", "") 
+        |> shared_test.withInputMapping("pay-ci", "src")
+      new InParallelStep {
+        in_parallel = new Listing<Step> {
+          shared_test.runCodeBuild("run-card-e2e-tests", "card.json", 1)
+          shared_test.runCodeBuild("run-products-e2e-tests", "products.json", 1)
+          shared_test.runCodeBuild("run-zap-e2e-tests", "zap.json", 1)
+        }
+      }
+      putStatus("pay-ci-endtoend-config-pull-request", "e2e tests", "success")
+    }
+  }
+}
+
+local function e2eJob(app: String, config: AppConfig) = new Job {
+  name = "\(app)-e2e"
+  // when (config.migrate_from.length > 0) { // TODO Remove after deployment
```

Comment:
Commented out code?

Reply:
Fixed. I commented that out while I was testing -- didn't want to rename the old jobs until the new ones were working properly.